stdlib/compare.ml

```diff
@@ -0,0 +1,37 @@
+(**************************************************************************)
+(*                                                                        *)
+(*                                 OCaml                                  *)
+(*                                                                        *)
+(*   Copyright 2025 Kate Deplaix                                          *)
+(*                                                                        *)
+(*   All rights reserved.  This file is distributed under the terms of    *)
+(*   the GNU Lesser General Public License version 2.1, with the          *)
+(*   special exception on linking described in the file LICENSE.          *)
+(*                                                                        *)
+(**************************************************************************)
+
+module Phys = struct
+  external equal : 'a -> 'a -> bool = "%eq"
+
+  module Infix = struct
```

Comment:
I prefer if there's a single name for these modules that redefine the meaning of your surface syntax. It makes it easier to spot their `open` which indicate that you need to be careful in what follows about the meaning of list constructors, indexing, infix, prefix and let operators.

In https://github.com/ocaml/ocaml/pull/2170 `Syntax` has been proposed for these modules. I was initially sceptical but I now think it's a good name (I have used it in various libs) and what I would prefer to see used here. It's a name that works for all kind of syntax redefinition and I don't think it's worth having more than one name for that or try to have more precise characterization of what is being redefined (the module prefix of `Syntax` usually gives a good hint).

Reply:
My impression from community libraries, was that `Syntax` was for `let+` type syntax and `Infix` was for the more traditional operators.

But sure, if the consensus is to have everything under `Syntax` in the standard library, i'm fine with that, although i find the presence of `return` in that submodule in #2170 confusing concerning the stated goal of such a module.

Reply:
Ah i see it was already mentioned in #2170 and seemed to be a consensus. I should've read the discussions before.

Anyway, done.